### PR TITLE
Rename autorange() to autorangeMode()

### PR DESCRIPTION
**Pull request by jtlan**
More accurately describes the functionality, and reduces confusion in the case of the getter:
```Javascript
plot.autorange(); // does this perform autoranging?
plot.autorangeMode(); // clearer
```
Also renamed the param to "autorangeMode".

### CODE REVIEW

**Comment by aicioara** [comment]
Links here https://github.com/palantir/plottable/issues/2272

File: src/components/plots/xyPlot.ts [hunk 1]
```diff
@@ -149,22 +149,22 @@ module Plottable {
...
      *   "x" will adjust the x Scale in relation to changes in the y domain.
      *   "y" will adjust the y Scale in relation to changes in the x domain.
      *   "none" means neither Scale will change automatically.
      * @returns {XYPlot} The calling XYPlot.
      */
-    public autorange(scaleName: string): XYPlot<X, Y>;
-    public autorange(scaleName?: string): any {
-      if (scaleName == null) {
+    public autorangeMode(mode: string): XYPlot<X, Y>;
+    public autorangeMode(mode?: string): any {
```
Comment (aicioara) [inline src/components/plots/xyPlot.ts]: `autorangeDirection` ? I don't have a strong preference though.

File: src/components/plots/xyPlot.ts [hunk 2]
```diff
@@ -149,22 +149,22 @@ module Plottable {
...
+     * @param {string} mode One of "x"/"y"/"none".
      *   "x" will adjust the x Scale in relation to changes in the y domain.
      *   "y" will adjust the y Scale in relation to changes in the x domain.
      *   "none" means neither Scale will change automatically.
      * @returns {XYPlot} The calling XYPlot.
      */
-    public autorange(scaleName: string): XYPlot<X, Y>;
-    public autorange(scaleName?: string): any {
-      if (scaleName == null) {
+    public autorangeMode(mode: string): XYPlot<X, Y>;
```
Comment (aicioara) [inline src/components/plots/xyPlot.ts]: Why not keep the parameter name same as the method name? We tend to do that in the rest of our codebase

File: src/components/plots/xyPlot.ts (hunk 1 above)
Comment (bluong) [inline src/components/plots/xyPlot.ts]: I like mode personally.

Can you rename the parameter name to `autorangeMode`?

File: src/components/plots/xyPlot.ts (hunk 2 above)
Comment (bluong) [inline src/components/plots/xyPlot.ts]: Agreed

**Comment by aicioara** [comment]
I'd say [NOQE] as it is straightforward single method rename

**Comment by aicioara** [comment]
Fix overlaying tests

**Pull request merged**